data/sql/updates/pending_db_world/rev_1558652406081124000.sql

```diff
@@ -0,0 +1,65 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1558652406081124000');
+
+-- Define helper functions
+
+DROP FUNCTION IF EXISTS InsertStrangeOreNode;
+
+DELIMITER $$
+
+CREATE FUNCTION InsertStrangeOreNode(position_x double, position_y double, position_z double)
+RETURNS decimal
+DETERMINISTIC
+BEGIN
+	SET @RND_ORIENTATION := (RAND() / 3.3) * 10;
```

Comment:
Could this be solved in another way? It means that each server installation has different values in the gameobject table, which is not good in my opinion.

Reply:
I could generate however many random numbers we need and just hardcode them, if you'd prefer.

So instead of

```SQL
...
CREATE FUNCTION InsertStrangeOreNode(position_x double, position_y double, position_z double)
RETURNS decimal
DETERMINISTIC
BEGIN
	SET @RND_ORIENTATION := (RAND() / 3.3) * 10;
	
	INSERT INTO `gameobject`(`id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`, `spawntimesecs`, `animprogress`, `state`, `ScriptName`, `VerifiedBuild`) VALUES
	(188699, 571, 0, 0, 1, 1, position_x, position_y, position_z, @RND_ORIENTATION, 0, 0, 0, 0, 300, 100, 1, '', 0);

	SET @NEW_GUID := LAST_INSERT_ID();

	RETURN @NEW_GUID;
END $$

DELIMITER ;

-- Remove old Strange Ore nodes

DELETE FROM `gameobject` where `id` = 188699 and `map` = 571;

-- Insert Strange Ore nodes

SET @rc := InsertStrangeOreNode(3898.41, -874.94, 109.69);
SET @rc := InsertStrangeOreNode(3910.97, -868.05, 107.67);
SET @rc := InsertStrangeOreNode(3939.70, -869.39, 104.37);
...
```

It could be

```SQL
...
CREATE FUNCTION InsertStrangeOreNode(position_x double, position_y double, position_z double, orientation double)
RETURNS decimal
DETERMINISTIC
BEGIN
	INSERT INTO `gameobject`(`id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`, `spawntimesecs`, `animprogress`, `state`, `ScriptName`, `VerifiedBuild`) VALUES
	(188699, 571, 0, 0, 1, 1, position_x, position_y, position_z, orientation, 0, 0, 0, 0, 300, 100, 1, '', 0);

	SET @NEW_GUID := LAST_INSERT_ID();

	RETURN @NEW_GUID;
END $$

DELIMITER ;

-- Remove old Strange Ore nodes

DELETE FROM `gameobject` where `id` = 188699 and `map` = 571;

-- Insert Strange Ore nodes

SET @rc := InsertStrangeOreNode(3898.41, -874.94, 109.69, 3.5);
SET @rc := InsertStrangeOreNode(3910.97, -868.05, 107.67, 1.6);
SET @rc := InsertStrangeOreNode(3939.70, -869.39, 104.37, 2.5);
...
```

It was just cheesy and obvious when *all* of the nodes were inserted in the same orientation. :)

Reply:
This would be much better. The DB content should be the same for all installations in order to make it easier to track down errors.

Could you please push a commit containing these changes?

Reply:
OK, changed, tested, and pushed.

Reply:
Great, thanks.